src/gccrs/args.rs

```diff
@@ -72,12 +72,21 @@ fn format_output_filename(
     Ok((output_file, crate_type))
 }
 
+/// Add `.tmp.o` to the expected output filename. Since we will already have produced the
+/// expected filename at this point, and we are likely currently converting it to a String
+/// to spawn as an argument, this function can avoid taking a Path as parameter and returning
+/// a PathBuf.
+fn object_file_name(output_file: &str) -> String {
+    format!("{}.tmp.o", output_file)
+}
+
 /// Structure used to represent arguments passed to `gccrs`. Convert them from `rustc`
 /// arguments using [`GccrsArg::from_rustc_arg`]
 pub struct GccrsArgs {
```

Comment:
Nit, i am not sure about the GccrsArgs object. To me this impl and structure represents 3 things:

1. Convert arguments to gccrs args
2. Create extra invocations to call ar to create static libs for example
3. represent Paths

I think it is outside of scope for this PR to change it, but it would like to see this refactored. Lets make an issue to plan it.